lib/widgets/amount_form_field.dart

```diff
@@ -70,7 +70,7 @@ class AmountFormField extends TextFormField {
             controller: controller,
             inputFormatters: accountModel.currency != Currency.SAT
                 ? [FilteringTextInputFormatter.allow(RegExp(r'\d+\.?\d*'))]
-                : [FilteringTextInputFormatter.digitsOnly],
+                : [SatAmountFormFieldFormatter()],
```

Comment:
Now that the formatter is changed the value contains spaces and when the validator tries to validate the form value here: https://github.com/breez/breezmobile/blob/5442a90e03426777acfee380bdf20e92e5303313/lib/widgets/amount_form_field.dart#L87
it will fail. You can reproduce that by pressing "CREATE" on the invoice page.
The failure is since the `currency.parse` expects only numbers.
It seems reasonable to me that we will change the `currency.parse` (for sats) to ignore spaces so the whole behaviour is symmetric.

Reply:
Don't worry, it is always better to fix things when they are fresh in mind, and you catch really good points as I have not test the rest of the flow 🤦 

Do you think is it ok applying the same regex `RegExp(r'[^\d*]')` on currency parse? So it removes the space and any other possible invalid thing, maybe a `sat` suffix.

Reply:
I tend to be more conservative here as the `parse` is used all over and there are many potential input sources...
Since we only format numbers and spaces it seems reasonable to me on parsing to ignore only spaces and leave the numbers.
Does it make sense to you? Or you have other valid cases in mind?

Reply:
Makes sense @roeierez I'm applying the changes :)

Reply:
Just made it, I add some tests to avoid future problems :)